### ISSUE TITLE
Issue when trying to join a singleplayer world

### ISSUE THREAD

**Issue by MenacingPerson** [PlanetTeamSpeakk/MoreCommands#26]
MC Version: 1.16.5
When i try to join singleplayer it crashes.
```---- Minecraft Crash Report ----
// Who set us up the TNT?

Time: 10/6/21 7:35 PM
Description: Starting integrated server

org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at Not Enough Crashes deobfuscated stack trace.(1.16.5+build.9)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:150)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at net.minecraft.server.integrated.IntegratedServer.<init>(IntegratedServer:52)
	at net.minecraft.client.MinecraftClient.method_29603(MinecraftClient:1853)
	at net.minecraft.class_310$$Lambda$6895/850209765.apply(Unknown Source)
	at net.minecraft.server.MinecraftServer.startServer(MinecraftServer:260)
	at net.minecraft.client.MinecraftClient.startIntegratedServer(MinecraftClient:1853)
	at net.minecraft.client.MinecraftClient.startIntegratedServer(MinecraftClient:1748)
	at net.minecraft.client.gui.screen.world.WorldListWidget$Entry.start(WorldListWidget:407)
	at net.minecraft.client.gui.screen.world.WorldListWidget$Entry.play(WorldListWidget:317)
	at net.minecraft.client.gui.screen.world.WorldListWidget$Entry.mouseClicked(WorldListWidget:254)
	at net.minecraft.client.gui.widget.EntryListWidget.mouseClicked(EntryListWidget:345)
	at net.minecraft.client.gui.ParentElement.mouseClicked(ParentElement:27)
	at net.minecraft.client.Mouse.method_1611(Mouse:92)
	at net.minecraft.class_312$$Lambda$4722/376673533.run(Unknown Source)
	at net.minecraft.client.gui.screen.Screen.wrapScreenError(Screen:435)
	at net.minecraft.client.Mouse.onMouseButton(Mouse:92)
	at net.minecraft.client.Mouse.method_22686(Mouse:162)
	at net.minecraft.class_312$$Lambda$4526/1974407064.run(Unknown Source)
	at net.minecraft.util.thread.ThreadExecutor.execute(ThreadExecutor:94)
	at net.minecraft.client.Mouse.method_22684(Mouse:162)
	at net.minecraft.class_312$$Lambda$3711/699892186.invoke(Unknown Source)
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3174)
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem:127)
	at net.minecraft.client.MinecraftClient.render(MinecraftClient:1071)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:681)
	at net.minecraft.client.main.Main.main(Main:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Constant modifier method processBlockBreakingAction_maxReach(D)D in morecommands.reach_mixins.json:common.MixinServerPlayerInteractionManager failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap MoreCommands-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:408)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1291)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1042)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351)
	... 40 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Starting integrated server --
Details:
	Level ID: Blanked out
	Level Name: Blanked out
Stacktrace:
	at net.minecraft.class_310.method_29610(class_310.java:1862)
	at net.minecraft.class_310.method_29606(class_310.java:1748)
	at net.minecraft.class_528$class_4272.method_20174(class_528.java:407)
	at net.minecraft.class_528$class_4272.method_20164(class_528.java:317)
	at net.minecraft.class_528$class_4272.method_25402(class_528.java:254)
	at net.minecraft.class_350.method_25402(class_350.java:345)
	at net.minecraft.class_4069.method_25402(class_4069.java:27)
	at net.minecraft.class_312.method_1611(class_312.java:92)
	at net.minecraft.class_312$$Lambda$4722/376673533.run(Unknown Source)
	at net.minecraft.class_437.method_25412(class_437.java:435)
	at net.minecraft.class_312.method_1601(class_312.java:92)
	at net.minecraft.class_312.method_22686(class_312.java:162)
	at net.minecraft.class_312$$Lambda$4526/1974407064.run(Unknown Source)
	at net.minecraft.class_1255.execute(class_1255.java:94)
	at net.minecraft.class_312.method_22684(class_312.java:162)
	at net.minecraft.class_312$$Lambda$3711/699892186.invoke(Unknown Source)
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3174)
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:127)
	at net.minecraft.class_310.method_1523(class_310.java:1071)
	at net.minecraft.class_310.method_1514(class_310.java:681)
	at net.minecraft.client.main.Main.main(Main.java:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
-- Affected screen --
Details:
	Screen name: net.minecraft.class_526
-- System Details --
Details:
	Minecraft Version: 1.16.5
	Minecraft Version ID: 1.16.5
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 3453296768 bytes (3293 MB) / 4294967296 bytes (4096 MB) up to 5737807872 bytes (5472 MB)
	CPUs: 8
	JVM Flags: 10 total; -Xmn128M -Xmx5455M -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Suspected Mods: Fabric Loader (fabricloader)
	Fabric Mods: 
		architectury: Architectury 1.17.16
		artifice: Artifice 0.14.6+1.16.5
		autoconfig1u: Auto Config v1 Updated 3.3.1
		betterbeds: Better Beds 1.1.0
		betterdroppeditems: Better Dropped Items 1.2.4-1.16.2
		betterf3: BetterF3 1.1.3
		blame: Blame 1.16.5-3.0.1-fabric
		blur: Blur (Fabric) 2.1.0
		carpet: Carpet Mod 1.4.38
		carpet-extra: Carpet Extra 1.4.34
		chime: Chime 1.1.3
		cloth-basic-math: Cloth Basic Math 0.5.1
		cloth-client-events-v0: Cloth Client Events v0 1.5.47
		cloth-config2: Cloth Config v4 4.11.26
		com_moandjiezana_toml_toml4j: toml4j 0.7.2
		connected_block_textures: Connected Block Textures 0.1.3+1.16
		craftpresence: CraftPresence 1.8.1
		dark-loading-screen: Dark Loading Screen 1.6.3
		dynamicfps: Dynamic FPS 2.0.2
		dynamicsoundfilters: Dynamic Sound Filters 1.2.0+1.16.5
		dynmus: Dynamic Music 1.2.0
		entityculling: EntityCulling-Fabric 1.2.2
		fabric: Fabric API 0.34.6+1.16
		fabric-api-base: Fabric API Base 0.3.0+c88702897d
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.0.0+dc716ea17d
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.10+3661eaa17d
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.5+ca58154a7d
		fabric-command-api-v1: Fabric Command API (v1) 1.1.1+351679a77d
		fabric-commands-v0: Fabric Commands (v0) 0.2.2+ca58154a7d
		fabric-containers-v0: Fabric Containers (v0) 0.1.11+9354966b7d
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.1+ca58154a7d
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.3+ca58154a7d
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.6+61336ba97d
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.1.0+c88702897d
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.2+ca58154a7d
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.1+ca58154a7d
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.6+ca58154a7d
		fabric-item-api-v1: Fabric Item API (v1) 1.2.1+ca58154a7d
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.7+2868a2287d
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.4+9354966b7d
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.1+ca58154a7d
		fabric-language-kotlin: Fabric Language Kotlin 1.3.72+build.1
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.2.1+ca58154a7d
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.2+ca58154a7d
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.3+ca58154a7d
		fabric-models-v0: Fabric Models (v0) 0.3.0+bc7a746f7d
		fabric-mumblelink-mod: MumbleLink 0.7.5
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.3+e3c9d0627d
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.8+ca58154a7d
		fabric-networking-v0: Fabric Networking (v0) 0.3.2+ca58154a7d
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.9.5+c88702897d
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.2+ca58154a7d
		fabric-particles-v1: Fabric Particles (v1) 0.2.4+ca58154a7d
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.4+ca58154a7d
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.2+2868a2287d
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.4+ca58154a7d
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.3.0+2868a2287d
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.5+ca58154a7d
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.13+ca58154a7d
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.2+ca58154a7d
		fabric-rendering-v1: Fabric Rendering (v1) 1.6.0+2868a2287d
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.6+2868a2287d
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.0+c045166c7d
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.5+9354966b7d
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.4+ca58154a7d
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.1+ca58154a7d
		fabric-textures-v0: Fabric Textures (v0) 1.0.6+ca58154a7d
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.7+8183d3957d
		fabricloader: Fabric Loader 0.11.3
		fabricskyboxes: FabricSkyBoxes 0.5.0
		fastfurnace: FastFurnace 2.4
		ferritecore: FerriteCore 2.0.4
		fiber: fiber 0.23.0-2
		hydrogen: Hydrogen 0.2
		indium: Indium 1.0.0+rev.0ad607b
		itemmodelfix: Item Model Fix 1.0.1
		java: Java HotSpot(TM) 64-Bit Server VM 8
		kirin: Kirin UI 1.7-1.16.2
		kronhud: KronHUD 1.1.1
		lambdabettergrass: LambdaBetterGrass 1.0.3+1.16
		lambdynlights: LambDynamicLights 1.3.4+1.16
		lazydfu: LazyDFU 0.1.2
		lithium: Lithium 0.6.4
		mcpatcherpatcher: MCPatcherPatcher 1.0.0+1.16.4
		minecraft: Minecraft 1.16.5
		mm: Manningham Mills 2.1
		modmenu: Mod Menu 1.16.9
		morecommands: MoreCommands 3.7
		no-potion-offset: No Potion Offset ${version}
		nofade: No Fade 1.16.5-1.0.1
		notenoughanimations: NotEnoughAnimations 1.2.4
		notenoughcrashes: Not Enough Crashes 3.2.0
		okzoomer: Ok Zoomer 4.0.1+1.16.2
		org_aperlambda_lambdajcommon: lambdajcommon 1.8.1
		org_codehaus_groovy_groovy: groovy 3.0.3
		org_codehaus_groovy_groovy-jsr223: groovy-jsr223 3.0.3
		org_jetbrains_annotations: annotations 17.0.0
		org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.3.72
		org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.3.72
		org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.3.72
		org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.3.72
		org_jetbrains_kotlinx_kotlinx-coroutines-core: kotlinx-coroutines-core 1.3.7
		org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.3.7
		pehkui: Pehkui 2.1.0+21w18a
		perspectivemod: Perspective Mod Redux 0.0.5
		phosphor: Phosphor 0.7.2+build.12
		pling: Pling 1.3.0
		polish: Polish 1.0.3
		presencefootsteps: Presence Footsteps r29+1.16.4
		replaymod: Replay Mod 1.16.4-2.5.2-67-gf5af283
		satin: Satin 1.5.1
		slight-gui-modifications: 'Slight' GUI Modifications 1.7.1
		smoothboot: Smooth Boot 1.16.5-1.6.0
		smoothscrollingeverywhere: Smooth Scrolling Everywhere 3.0.3-unstable
		sodium: Sodium 0.2.0+unknown
		someforgepatchesported: Some Forge Patches Ported 1.0.0
		spruceui: SpruceUI 2.0.4+1.16
		terra: Terra 5.3.3-BETA+5dd00db8
		timetolive: Time To Live 0.3
		transliterationlib: TRansliterationLib 1.0.4
		worldedit: WorldEdit 7.2.5+57d5ac9
		worldeditcui: WorldEditCUI 1.16.5+02
	Launched Version: Fabric 1.16.5
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: Intel(R) UHD Graphics GL version 4.6.0 - Build 27.20.100.9466, Intel
	GL Caps: Using framebuffer using OpenGL 3.0
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Graphics mode: fast
	Resource Packs: replaymod_lang (incompatible), vanilla, Fabric Mods, slightguimodifications:cts_textures, betterbeds/connectedbeds, betterbeds/fancybeds, lambdabettergrass/default, file/Compliance-32x-Java-Beta-10.zip
	Current Language: English (US)
	CPU: 8x Intel(R) Core(TM) i5-1035G1 CPU @ 1.00GHz
	Client Crashes Since Restart: 1
	Integrated Server Crashes Since Restart: 0
```
Doesn't crash without this mod 🤷

**Comment by MenacingPerson** [comment]
Is nobody gonna reply? ok

**Comment by MenacingPerson** [comment]
still no reply... can i help in any way whatsoever? If I try with this mod and fapi, it makes all textures messed up.

**Comment by PlanetTeamSpeakk** [comment]
Seeing as I cannot replicate this issue myself, I can't really fix it.
Can you ensure you use the latest version of the Fabric Loader, Fabric API and MoreCommands?
Also last question, do you get this error when only using MoreCommands and Fabric API or are you using other mods alongside it?

**Comment by MenacingPerson** [comment]
ah, thanks for the response. I will try it again with a new version of MoreCommands.

**Comment by MenacingPerson** [comment]
Still crashes, Fabric api 0.35.0, Fabric loader 0.11.3, MoreCommands 3.7.2, Minecraft 1.16.5. There is a list of mods in the crash log.
https://hastebin.com/betobehoja.yaml

**Comment by MenacingPerson** [comment]
Will try with fabric api 0.36.0 and loader 0.11.6

**Comment by Samisaprogrammer** [comment]
I meant fabric api 0.36.0

**Comment by MenacingPerson** [comment]
Still crashes. https://hastebin.com/xonisahesi.yaml

**Comment by MenacingPerson** [comment]
Only crashes when trying a singleplayer world.

**Comment by MenacingPerson** [comment]
Will try with only fabric api and fabric lang kotlin.

**Comment by MenacingPerson** [comment]
Doesn't crash, so definitely a mod incompatibility.

**Comment by PlanetTeamSpeakk** [comment]
Could you try to isolate what mod is causing the compatibility issue so I can try to fix it?

**Comment by MenacingPerson** [comment]
Oh! It's pehkui that is causing the incompatibility, probably because of the /reach command!

**Comment by MenacingPerson** [comment]
A suggestion is that you could use playerabilitylib, although idk if pehkui even supports it.

**Comment by PlanetTeamSpeakk** [comment]
Has been fixed in 3.8.1.
